### PR TITLE
mitigate performance issue with userscope

### PR DESCRIPTION
**Pull request by jbreiding**
**What changed?**
changes to function receivers for possible performance issue

**Why?**
because throughput dropped!

**How did you test it?**
dev environment testing

**Potential risks**
None

**Is hotfix candidate?**
No

### CODE REVIEW

**Comment by jbreiding** [comment]
Leaving this open during troubleshooting

File: common/metrics/events_scope.go [hunk 1]
```diff
@@ -182,13 +182,13 @@ func (e *eventsStopwatch) Subtract(d time.Duration) {
...
+		if m, ok = MetricDefs[Common][id]; !ok {
 			panic(fmt.Errorf("failed to lookup metric by id %v and service %v", id, idx))
 		}
-	}
 
-	return m
+		return m
+	} else {
+		return m
+	}
```
Comment (yiminc) [inline common/metrics/events_scope.go:193]: nit: why this change? seems the old one is more clear.

File: common/metrics/events_userscope.go [hunk 2]
```diff
@@ -66,27 +74,38 @@ func (e eventsUserScope) StartTimer(timer string) Stopwatch {
...
+	switch {
+	case tags != nil && e.tags == nil:
+		return newEventsUserScope(e.provider, tags)
+	case tags == nil && e.tags != nil:
+		return newEventsUserScope(e.provider, e.tags)
+	default:
+		m := maps.Clone(e.tags)
+		maps.Copy(m, tags)
+		return newEventsUserScope(e.provider, m)
+	}
```
Comment (yiminc) [inline common/metrics/events_userscope.go:108]: would it be better to check len(map) == 0?
And do we care the case both are empty?

File: common/metrics/events_userscope.go [hunk 3]
```diff
@@ -66,27 +74,38 @@ func (e eventsUserScope) StartTimer(timer string) Stopwatch {
...
 }
 
 func mapToTags(m map[string]string) []Tag {
```
Comment (yiminc) [inline common/metrics/events_userscope.go:111]: delete this?

Comment (jbreiding) [reply common/metrics/events_userscope.go:112]: before this PR is finalized this will either be used or deleted.

File: common/metrics/events_userscope.go [hunk 4]
```diff
@@ -34,62 +34,94 @@ import (
...
 // Tagged returns a new scope with added and/or overriden tags values that can be used
 // to provide additional information to metrics
-func (e eventsUserScope) Tagged(tags map[string]string) UserScope {
-	return newEventsUserScope(e.provider.WithTags(mapToTags(tags)...))
+func (e *eventsUserScope) Tagged(tags map[string]string) UserScope {
+	if len(tags) == 0 {
+		return newEventsUserScope(e.provider, e.tags)
+	}
+
+	if len(tags) != 0 && len(e.tags) == 0 {
```
Comment (yiminc) [inline common/metrics/events_userscope.go:111]: nit: no need to check len(tags) != 0

File: common/metrics/events_userscope.go (hunk 3 above)
Comment (jbreiding) [reply common/metrics/events_userscope.go:117]: function is being used now
